### PR TITLE
Feature/add user profile logic

### PR DESCRIPTION
**Pull request by tanyushanel**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"editor.defaultFormatter": "esbenp.prettier-vscode"
```
Comment (RomPavel) [inline .vscode/settings.json:2]: We need to add .vscode folder to .gitignore

File: exadel-trainee-app/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/app-routing.module.ts:1]: Please remove all this lines

File: exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -1,10 +1,124 @@
...
 })
 export class UserProfileComponent implements OnInit {
-  ngOnInit(): void {}
+  levels = Level;
+
+  selectedLevel = Level.beginner;
+
+  constructor(private router: Router) {}
+
+  results: Test[] = [
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts:18]: Lets put this array to mocks/test-mocks.ts

File: exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts [hunk 4]
```diff
@@ -1,10 +1,124 @@
...
+      },
+      feedback: 'OK',
+    },
+  ];
+
+  ngOnInit() {}
+
+  onStartButtonClick(): void {
+    this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
+      this.router.navigate([`grammar`]);
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-profile.component.ts:121]: Lets use Route enum here

File: .vscode/settings.json (hunk 1 above)
Comment (tanyushanel) [reply .vscode/settings.json:2]: image

Comment (tanyushanel) [reply .vscode/settings.json:2]: image

**Comment by tanyushanel** [comment]
image

File: exadel-trainee-app/src/constants/mock-test-data.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable prettier/prettier */
```
Comment (RomPavel) [inline exadel-trainee-app/src/constants/mock-test-data.ts:1]: remove this

File: exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
 import { UserProfileRoutingModule } from './user-profile-routing.module';
 import { UserProfileComponent } from './user-profile.component';
+import { UserResultsTableComponent } from './user-results-table/user-results-table.component';
 
 @NgModule({
-  declarations: [UserProfileComponent],
-  imports: [CommonModule, UserProfileRoutingModule],
+  declarations: [UserProfileComponent, UserResultsTableComponent],
+  imports: [CommonModule, UserProfileRoutingModule, AngularMaterialCommonModule],
+  exports: [UserResultsTableComponent],
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts:11]: Why do you need to export this table, you plan to use it inside another module?

File: exadel-trainee-app/src/app/models/user-profile/notification/notification.component.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-notification',
+  templateUrl: './notification.component.html',
+  styleUrls: ['./notification.component.scss'],
+})
+export class NotificationComponent implements OnInit {
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/notification/notification.component.ts:8]: Why do you have two NotificationComponent in different modules?

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.module.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.module.ts:1]: remove this

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 9]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:1]: remove this

File: exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.html [hunk 10]
```diff
@@ -12,7 +12,13 @@
   </button>
 
   <div *ngIf="results.length > 0; else noResults">
-    <app-dialog-results-table [results]="results"></app-dialog-results-table>
+    <!-- <app-dialog-results-table [results]="results"></app-dialog-results-table> -->
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.html:15]: do we need this comment?

File: exadel-trainee-app/src/app/models/dialog-module/common-dialog.module.ts [hunk 11]
```diff
@@ -1,15 +1,12 @@
...
 import { UserResultsDialogComponent } from './user-results-dialog/user-results-dialog.component';
-import { DialogResultsTableComponent } from './user-results-dialog/dialog-results-table/dialog-results-table.component';
+import { AngularMaterialCommonModule } from '../angular-material-common.module';
+import { UserProfileModule } from '../user-profile/user-profile.module';
 
 @NgModule({
-  declarations: [UserResultsDialogComponent, DialogResultsTableComponent],
-  imports: [CommonModule, MatDialogModule, MatButtonModule, MatCardModule, MatTableModule],
+  declarations: [UserResultsDialogComponent],
+  imports: [CommonModule, AngularMaterialCommonModule, UserProfileModule],
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/dialog-module/common-dialog.module.ts:9]: Why you import whole **UserProfileModule** in some common module? 

- [ ] remove it

File: exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts (hunk 6 above)
Comment (tanyushanel) [reply exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts:11]: Yes, in modal for HR

File: exadel-trainee-app/src/app/models/user-profile/notification/notification.component.ts (hunk 7 above)
Comment (tanyushanel) [reply exadel-trainee-app/src/app/models/user-profile/notification/notification.component.ts:8]: One is for deleting. In userProfile. Miss to delete.

File: exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.html (hunk 10 above)
Comment (tanyushanel) [reply exadel-trainee-app/src/app/models/dialog-module/user-results-dialog/user-results-dialog.component.html:15]: No )

File: exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts (hunk 6 above)
Comment (RomPavel) [reply exadel-trainee-app/src/app/models/user-profile/user-profile.module.ts:11]: Then this table should be in separate small module that should be imported here and in hr module

File: exadel-trainee-app/src/app/models/dialog-module/common-dialog.module.ts (hunk 11 above)
Comment (tanyushanel) [reply exadel-trainee-app/src/app/models/dialog-module/common-dialog.module.ts:9]: Reuse a table from UserProfileModule in dialog for hrProfile

File: exadel-trainee-app/src/app/models/user-profile/user-results-table/user-results-table.component.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+      state('collapsed', style({ height: '0px', minHeight: '0' })),
+      state('expanded', style({ height: '*' })),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
+    ]),
+  ],
+})
+export class UserResultsTableComponent implements OnInit {
+  @Input() results: Test[] = [];
+
+  columnsToDisplay: any[] = ['id', 'date', 'level', 'result'];
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/user-profile/user-results-table/user-results-table.component.ts:20]: its a **string[]** not **any[]**
Its a bad practice to use any

**Comment by tanyushanel** [comment]
image

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 13]
```diff
@@ -38,9 +38,10 @@ export class HrProfileComponent implements AfterViewInit {
     }
   }
 
-  onOpenUserResultsDialog(): void {
+  onOpenUserResultsDialog(row: any): void {
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:41]: Dont use any

**Review by RomPavel — APPROVED** [review]

**Pull request merged**